zulip/zulip/cli.py

```diff
@@ -0,0 +1,195 @@
+            "type": "private",
+            "content": message,
+            "to": recipients,
+        }
+
+    if message_data["type"] == "stream":
+        log.info(
+            'Sending message to stream "%s", subject "%s"... '
+            % (message_data["to"], message_data["subject"])
+        )
+    else:
+        log.info("Sending message to %s... " % message_data["to"])
+    response = client.send_message(message_data)
+    log_exit(response)
+
+
+@cli.command()
+def upload_file() -> None:
+    """Upload a single file and get the corresponding URI."""
+    # TODO
+
+
+@cli.command()
+@click.argument("message_id", type=int)
+@click.option("--message", "-m", required=True)
+def update_message(message_id: int, message: str) -> None:
+    """Edit/update the content or topic of a message."""
+    request = {
+        "message_id": message_id,
+        "content": message,
+    }
+    response = client.update_message(request)
+    log_exit(response)
+
+
+@cli.command()
+@click.argument("message_id", type=int)
+def delete_message(message_id: int) -> None:
+    """Permanently delete a message."""
+    response = client.delete_message(message_id)
+    log_exit(response)
+
+
+# TODO
+# https://zulip.com/api/get-messages
+# https://zulip.com/api/construct-narrow
+
+
+@cli.command()
+@click.argument("message_id", type=int)
+@click.argument("emoji_name")
+def add_reaction(message_id: int, emoji_name: str) -> None:
+    """Add an emoji reaction to a message."""
+    request = {
+        "message_id": message_id,
+        "emoji_name": emoji_name,
+    }
+
+    response = client.add_reaction(request)
+    log_exit(response)
+
+
+@cli.command()
+@click.argument("message_id", type=int)
+@click.argument("emoji_name")
+def remove_reaction(message_id: int, emoji_name: str) -> None:
+    """Remove an emoji reaction from a message."""
+    request = {
+        "message_id": message_id,
+        "emoji_name": emoji_name,
+    }
+
+    response = client.remove_reaction(request)
+    log_exit(response)
+
+
+# TODO
+# https://zulip.com/api/render-message
+# https://zulip.com/api/get-raw-message
+# https://zulip.com/api/check-narrow-matches
+
+
+@cli.command()
+@click.argument("message_id", type=int)
+def get_message_history(message_id: int) -> None:
+    """Fetch the message edit history of a previously edited message.
+    Note that edit history may be disabled in some organizations; see https://zulip.com/help/view-a-messages-edit-history.
+    """
+    response = client.get_message_history(message_id)
+    log_exit(response)
+
+
+# TODO
+# https://zulip.com/api/update-message-flags
+
+
+@cli.command()
+def mark_all_as_read() -> None:
+    """Marks all of the current user's unread messages as read."""
+    response = client.mark_all_as_read()
+    log_exit(response)
+
+
+# Streams API
+
+
+@cli.command()
+def get_subscriptions() -> None:
+    """Get all streams that the user is subscribed to."""
+    # TODO python-zulip-api deviates from zulip.yaml here
```

Comment:
I think python-zulip-api is wrong, and we should migrate it.  

Can you open an issue or quick PR?

Reply:
This comment is obsolete. I already made the PR in #665. Already merged.